mods/page_revisions.js

```diff
@@ -335,14 +448,33 @@ PRS.prototype.fetchAndStoreMWRevision = function(restbase, req) {
 PRS.prototype.getTitleRevision = function(restbase, req) {
     var self = this;
     var rp = req.params;
+    var title = rbUtil.normalizeTitle(rp.title);
     var revisionRequest;
+    var pageDataRequest = restbase.get({
+        uri: self.pageTableURI(rp.domain),
+        body: {
+            table: self.pageTableName,
+            attributes: {
+                title: title
+            }
```

Comment:
Since undeletions should be rather rare compared to regular getTitleRevision requests, I think it's worth optimizing this. The difference should be significant for pages with a couple of entries.